android/java/org/chromium/chrome/browser/upgrade/BraveUpgradeJobIntentService.java

```diff
@@ -0,0 +1,202 @@
+
+            // The shields_config.dat file strips "www." from
+            // hostnames, so include it back (with scheme!) to avoid
+            // missing sites
+            if (!hasSubdomain(host)) {
+                migrateShieldsSettingsForHost("http://www." + host, settings);
+                migrateShieldsSettingsForHost("https://www." + host, settings);
+            }
+        }
+
+        return true;
+    }
+
+    private boolean migrateShieldsSettingsForHost(String host, String settings) {
+        String[] array = new String(settings).split(",");
+        if (array.length < 6) {
+            Log.e(TAG, "Brave shields host settings invalid for host " + host);
+            return false;
+        }
+
+        Profile profile = Profile.getLastUsedProfile();
+        BraveShieldsContentSettings.setShields(profile, host,
+                BraveShieldsContentSettings.RESOURCE_IDENTIFIER_BRAVE_SHIELDS, array[0].equals("1"),
+                false);
+        BraveShieldsContentSettings.setShields(profile, host,
+                BraveShieldsContentSettings.RESOURCE_IDENTIFIER_ADS_TRACKERS, array[1].equals("1"),
+                false);
+        BraveShieldsContentSettings.setShields(profile, host,
+                BraveShieldsContentSettings.RESOURCE_IDENTIFIER_HTTP_UPGRADABLE_RESOURCES,
+                array[2].equals("1"), false);
+        BraveShieldsContentSettings.setShields(profile, host,
+                BraveShieldsContentSettings.RESOURCE_IDENTIFIER_JAVASCRIPTS, array[3].equals("1"),
+                false);
+        BraveShieldsContentSettings.setShields(profile, host,
+                BraveShieldsContentSettings.RESOURCE_IDENTIFIER_COOKIES, array[4].equals("1"),
+                false);
+        BraveShieldsContentSettings.setShields(profile, host,
+                BraveShieldsContentSettings.RESOURCE_IDENTIFIER_FINGERPRINTING,
+                array[5].equals("1"), false);
+
+        return true;
+    }
+
+    @Override
+    protected void onHandleWork(Intent intent) {
+        // Kick off the migration task only after the browser has
+        // completed startup, as migration requires a profile.
+        PostTask.runOrPostTask(UiThreadTaskTraits.DEFAULT, () -> {
+            BrowserStartupController.get(LibraryProcessType.PROCESS_BROWSER)
+                    .addStartupCompletedObserver(new BrowserStartupController.StartupCallback() {
+                        @Override
+                        public void onSuccess() {
+                            if (!migrateShieldsConfig()) {
+                                Log.e(TAG, "Failed to migrate Brave shields config settings");
+                                return;
+                            }
+
+                            Log.i(TAG, "Successfully migrated Brave shields config settings");
+
+                            // Rename the migrated file so we don't
+                            // try to import it again
```

Comment:
Do we keep the migrated file forever? It's fine as to me, in case of a failures we can rerun the task in the future.

Reply:
Yes, we keep it forever. Was playing it safe, just in case there was a problem during migration.